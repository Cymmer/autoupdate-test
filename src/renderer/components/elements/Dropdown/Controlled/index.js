import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Select from 'react-select';
import GLOBALS from 'codechum-app-globals';
import RequiredField from 'components/elements/RequiredField';
import styles from '../styles.module.scss';

import Text from '../../Text';
import textTypes from '../../Text/constants/textTypes';

import { getIsNightMode } from '../../../../ducks';

import dropdownTypes from '../constants/dropdownTypes';
import determineStyles from '../styles/determineStyles';

import CustomOption from '../custom/Option';
import CustomSingleValue from '../custom/SingleValue';
import CustomValueContainer from '../custom/ValueContainer';

/* eslint-disable react/display-name */
const ControlledDropdown = ({
  options,
  isMulti,
  name,
  value,
  error,
  onChange,
  placeholder,
  isRequired,
  helperText,
  className,
  type,
  label,
  isNightMode,
  isClearable,
  disabled,
}) => (
  <div className={cn(className, styles.Dropdown)} id={name}>
    {label && type === dropdownTypes.SLIM && (
      <Text
        className={styles.Dropdown___slim_label}
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
        data-test="dropdownLabel"
        element="label"
        id="dropdownLabel"
        type={textTypes.BODY.MD}
      >
        {isRequired ? <RequiredField placeholder={label} /> : label}
      </Text>
    )}
    <Select
      components={{
        Option: CustomOption,
        SingleValue: CustomSingleValue,
        ValueContainer: (valueContainerProps) => (
          <CustomValueContainer type={type} {...valueContainerProps} />
        ),
        IndicatorSeparator: null,
      }}
      isClearable={isClearable}
      isDisabled={disabled}
      isMulti={isMulti}
      name={name}
      options={options}
      placeholder={
        isRequired ? <RequiredField placeholder={placeholder} /> : placeholder
      }
      styles={determineStyles(type, isNightMode)}
      value={value}
      onChange={onChange}
    />
    {(helperText || error) && (
      <div className={styles.Dropdown_helperTextContainer}>
        {helperText && (
          <Text
            className={styles.Dropdown_helperText}
            colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
            type={textTypes.BODY.XS}
          >
            {helperText}
          </Text>
        )}
        {error && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.RED['500']}
            data-test="inputError"
            type={textTypes.BODY.XS}
          >
            {error}
          </Text>
        )}
      </div>
    )}
  </div>
);

ControlledDropdown.defaultProps = {
  className: null,
  label: null,
  type: dropdownTypes.FORM,
  value: null,
  isMulti: false,
  placeholder: null,
  helperText: null,
  error: null,
  isClearable: false,
  disabled: false,
};

ControlledDropdown.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.oneOf([
    dropdownTypes.FORM,
    dropdownTypes.SLIM,
    dropdownTypes.PLAYGROUND,
    dropdownTypes.LARGE,
  ]),
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.oneOfType([
    // Proptype for the non-grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
          .isRequired,
        label: PropTypes.string.isRequired,

        // custom icon in each option (a custom component)
        icon: PropTypes.element,
      })
    ),

    // Proptype for the grouped options
    PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        options: PropTypes.arrayOf(
          PropTypes.shape({
            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
              .isRequired,
            label: PropTypes.string.isRequired,

            // custom icon in each option (a custom component)
            icon: PropTypes.element,
          })
        ),
      })
    ),
  ]).isRequired,
  isMulti: PropTypes.bool,
  value: PropTypes.oneOfType([
    PropTypes.shape({
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
      label: PropTypes.string.isRequired,

      // custom icon in each option (a custom component)
      icon: PropTypes.element,
    }),
    PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
          .isRequired,
        label: PropTypes.string.isRequired,

        // custom icon in each option (a custom component)
        icon: PropTypes.element,
      })
    ),
  ]),
  placeholder: PropTypes.string,
  helperText: PropTypes.string,
  isNightMode: PropTypes.bool.isRequired,
  isClearable: PropTypes.bool,
  disabled: PropTypes.bool,
  isRequired: PropTypes.bool,
};

const mapStateToProps = (store) => ({
  isNightMode: getIsNightMode(store),
});

export default connect(mapStateToProps, null)(ControlledDropdown);
