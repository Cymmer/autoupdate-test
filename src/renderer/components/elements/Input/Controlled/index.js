import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import RequiredField from 'components/elements/RequiredField';
import styles from '../styles.module.scss';

import Text from '../../Text';
import textTypes from '../../Text/constants/textTypes';
import Icon from '../../Icon';

import inputKinds from '../constants/inputKinds';
import inputTypes from '../constants/inputTypes';

const ControlledInput = ({
  id,
  type,
  kind,
  className,
  placeholder,
  isRequired,
  error,
  success,
  name,
  label,
  disabled,
  value,
  onChange,
  icon,
  maxLength,
  autoComplete,
  helperText,
  step,
  tabIndex,
  min,
  max,
}) => (
  <div className={cn(className, styles.Input_container)}>
    {label && type !== inputTypes.FORM && (
      <Text
        className={styles[`Input___${type}_label`]}
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
        element="label"
        type={textTypes.BODY.MD}
      >
        {isRequired ? <RequiredField placeholder={label} /> : label}
      </Text>
    )}

    <input
      autoComplete={autoComplete}
      className={cn(styles[`Input___${type}`], {
        [styles[`Input___${type}___icon`]]: icon !== null,
        [styles.Input___success]: success !== null,
        [styles.Input___error]: error !== null,
      })}
      data-test="input"
      disabled={disabled}
      id={id || name}
      max={max}
      maxLength={maxLength}
      min={min}
      name={name}
      placeholder={type !== inputTypes.FORM && placeholder ? placeholder : null}
      step={step}
      tabIndex={tabIndex}
      type={kind}
      value={value || ''}
      onChange={onChange}
    />

    {placeholder && type === inputTypes.FORM && (
      <Text
        className={cn(styles.Input___form_label, {
          [styles.Input___form_label___active]: value !== '',
        })}
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['500']}
        type={textTypes.BODY.MD}
      >
        {isRequired ? <RequiredField placeholder={placeholder} /> : placeholder}
      </Text>
    )}

    {icon && <Icon className={styles[`Input___${type}_icon`]} icon={icon} />}

    {(helperText || success || error) && (
      <div className={styles.Input_helperTextContainer}>
        {helperText && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['500']}
            type={textTypes.BODY.XS}
          >
            {helperText}
          </Text>
        )}

        {error && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.RED['500']}
            type={textTypes.BODY.XS}
          >
            {error}
          </Text>
        )}

        {success && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.GREEN['500']}
            type={textTypes.BODY.XS}
          >
            {success}
          </Text>
        )}
      </div>
    )}
  </div>
);

ControlledInput.defaultProps = {
  id: null,
  type: inputTypes.SLIM,
  kind: inputKinds.TEXT,
  className: null,
  placeholder: null,
  error: null,
  success: null,
  label: null,
  disabled: false,
  value: '',
  maxLength: null,
  autoComplete: null,
  helperText: null,
  icon: null,
  step: null,
  tabIndex: null,
  min: null,
  max: null,
};

ControlledInput.propTypes = {
  type: PropTypes.oneOf([inputTypes.FORM, inputTypes.SLIM, inputTypes.CODE]),
  kind: PropTypes.oneOf([
    inputKinds.NUMBER,
    inputKinds.PASSWORD,
    inputKinds.TEXT,
  ]),
  id: PropTypes.string,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  error: PropTypes.string,
  success: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  disabled: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  icon: PropTypes.string,
  maxLength: PropTypes.number,
  autoComplete: PropTypes.string,
  helperText: PropTypes.string,
  step: PropTypes.number,
  tabIndex: PropTypes.number,
  min: PropTypes.number,
  max: PropTypes.number,
  isRequired: PropTypes.bool,
};

export default ControlledInput;
