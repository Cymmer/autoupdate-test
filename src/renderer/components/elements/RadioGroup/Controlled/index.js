import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from '../radio.module.scss';

import Icon from '../../Icon';

import Label from '../../Label';

import radioTypes from '../constants/radioTypes';

const ControlledRadio = ({
  name,
  label,
  id,
  checked,
  className,
  onChange,
  disabled,
  value,
  type,
  icon,
}) => (
  <div
    className={cn(className, styles[`Radio___${type}`], {
      [styles[`Radio___${type}___disabled`]]: disabled,
    })}
    id={id ? `${id}Container` : null}
  >
    <input
      checked={checked}
      className={styles[`Radio___${type}_input`]}
      data-test="radio"
      disabled={disabled}
      id={id}
      name={name}
      type="radio"
      value={value}
      onChange={onChange}
    />
    <Label
      className={styles[`Radio___${type}_label`]}
      disabled={disabled}
      htmlFor={id}
    >
      {icon && type === radioTypes.LARGE && (
        <Icon className={styles.Radio___large_icon} icon={icon} />
      )}
      {label}
    </Label>
  </div>
);

ControlledRadio.defaultProps = {
  checked: false,
  className: null,
  disabled: false,
  type: radioTypes.SMALL,
  icon: null,
};

ControlledRadio.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.any.isRequired,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf([radioTypes.SMALL, radioTypes.LARGE]),
  icon: PropTypes.string,
};

export default ControlledRadio;
