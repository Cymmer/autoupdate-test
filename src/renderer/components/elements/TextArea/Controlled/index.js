import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import RequiredField from 'components/elements/RequiredField';
import styles from '../styles.module.scss';

import Text from '../../Text';
import textTypes from '../../Text/constants/textTypes';
import Icon from '../../Icon';

import textAreaTypes from '../constants/textAreaTypes';

const ControlledTextArea = ({
  id,
  type,
  className,
  inputClassName,
  placeholder,
  isRequired,
  error,
  success,
  name,
  disabled,
  value,
  icon,
  helperText,
  readOnly,
  onChange,
}) => (
  <div className={cn(className, styles.TextArea_container)}>
    <textarea
      className={cn(styles[`TextArea___${type}`], inputClassName, {
        [styles.TextArea___icon]: icon !== null,
        [styles.TextArea___success]: success !== null,
        [styles.TextArea___error]: error !== null,
      })}
      data-test="textArea"
      disabled={disabled}
      id={id}
      name={name}
      placeholder={type !== textAreaTypes.FORM ? placeholder : ''}
      readOnly={readOnly}
      value={value || ''}
      onChange={onChange}
    />

    {placeholder && type === textAreaTypes.FORM && (
      <Text
        className={cn(styles.TextArea___form_placeholder, {
          [styles.TextArea___form_placeholder___active]: value !== '',
        })}
        colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['500']}
        type={textTypes.BODY.MD}
      >
        {isRequired ? <RequiredField placeholder={placeholder} /> : placeholder}
      </Text>
    )}

    {icon && <Icon className={styles.TextArea_icon} icon={icon} />}

    {(helperText || success || error) && (
      <div className={styles.TextArea_helperTextContainer}>
        {helperText && (
          <Text
            className={styles.TextArea_helperText}
            colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['500']}
            type={textTypes.BODY.XS}
          >
            {helperText}
          </Text>
        )}

        {error && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.RED['500']}
            type={textTypes.BODY.XS}
          >
            {error}
          </Text>
        )}

        {success && (
          <Text
            colorClass={GLOBALS.COLOR_CLASSES.GREEN['500']}
            type={textTypes.BODY.XS}
          >
            {success}
          </Text>
        )}
      </div>
    )}
  </div>
);

ControlledTextArea.defaultProps = {
  id: null,
  type: textAreaTypes.FORM,
  className: null,
  inputClassName: null,
  placeholder: null,
  error: null,
  success: null,
  disabled: false,
  value: '',
  helperText: null,
  readOnly: false,
  icon: null,
};

ControlledTextArea.propTypes = {
  id: PropTypes.string,
  type: PropTypes.oneOf([textAreaTypes.CODE, textAreaTypes.FORM]),
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  placeholder: PropTypes.string,
  error: PropTypes.string,
  success: PropTypes.string,
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  icon: PropTypes.string,
  helperText: PropTypes.string,
  readOnly: PropTypes.bool,
  isRequired: PropTypes.bool,
};

export default ControlledTextArea;
