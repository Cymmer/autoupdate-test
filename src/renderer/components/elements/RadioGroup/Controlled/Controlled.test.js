import React from 'react';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import Controlled from './index';

describe('RENDERING - RadioGroup > Controlled', () => {
  it('should render with default props', () => {
    const component = shallow(
      <Controlled
        id="test"
        label="Test"
        name="test"
        value="test"
        onChange={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render an icon if it has an icon', () => {
    const component = shallow(
      <Controlled
        icon="test"
        id="test"
        label="Test"
        name="test"
        value="test"
        onChange={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe('EVENTS - RadioGroup > Controlled', () => {
  it('should be changeable if not disabled', () => {
    const changeFn = jest.fn();
    const component = mount(
      <Controlled
        id="test"
        label="Test"
        name="test"
        value="test"
        onChange={changeFn}
      />
    );

    const radio = findByTestAttr(component, 'radio');
    radio.simulate('change', {
      target: {
        checked: true,
      },
    });

    expect(changeFn).toHaveBeenCalled();
    expect(radio.prop('disabled')).toBe(false);
  });

  it('should not be changeable if disabled', () => {
    const changeFn = jest.fn();
    const component = mount(
      <Controlled
        disabled
        id="test"
        label="Test"
        name="test"
        value="test"
        onChange={changeFn}
      />
    );

    const radio = findByTestAttr(component, 'radio');

    // in checking for a radio if it is indeed disabled,
    // we do not simulate a change event because that
    // overrides the disabled state. Instead, we just check
    // if the disabled prop was indeed passed to the radio
    expect(radio.prop('disabled')).toBe(true);
  });
});
