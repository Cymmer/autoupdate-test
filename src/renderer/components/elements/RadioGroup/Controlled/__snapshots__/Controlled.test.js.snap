// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - RadioGroup > Controlled should render an icon if it has an icon 1`] = `
<div
  className="Radio___small"
  id="testContainer"
>
  <input
    checked={false}
    className="Radio___small_input"
    data-test="radio"
    disabled={false}
    id="test"
    name="test"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <Label
    className="Radio___small_label"
    disabled={false}
    htmlFor="test"
    id={null}
  >
    Test
  </Label>
</div>
`;

exports[`RENDERING - RadioGroup > Controlled should render with default props 1`] = `
<div
  className="Radio___small"
  id="testContainer"
>
  <input
    checked={false}
    className="Radio___small_input"
    data-test="radio"
    disabled={false}
    id="test"
    name="test"
    onChange={[Function]}
    type="radio"
    value="test"
  />
  <Label
    className="Radio___small_label"
    disabled={false}
    htmlFor="test"
    id={null}
  >
    Test
  </Label>
</div>
`;
